### PR TITLE
Associate .chn with chaiNNer on MacOS

### PR DESCRIPTION
**Pull request by stonerl**
- added association of .chn filetype with chainner on macOS
- added file type icon
- added open-file event for macOS

This addresses #1062 but does not solve it completely, since file association only works on macOS ATM.

### CODE REVIEW

**Comment by stonerl** [comment]
CleanShot 2023-08-18 at 12 05 32

**Comment by RunDevelopment** [comment]
Thank you @stonerl!

Let's focus on `.chn` files for now. Opening models presents the problem that the PyTorch, ONNX, and NCNN packages may not be installed, so what are we supposed to do then? Also, there are no `.ncnn` files. NCNN stores models as 2 files: `.param` and `.bin`.

**Comment by stonerl** [comment]
Okay, changed it. It now only associates chaiNNer with .chn files.

**Comment by RunDevelopment** [comment]
Do we still need to other icons?

**Comment by stonerl** [comment]
> Do we still need to other icons?

I would let them stay. The are not copied into the app, so no harm done. And if we decide to make use of them then we already have them.

**Comment by RunDevelopment** [comment]
Doesn't feel good to have unused files in the repository...

No, please remove them. Let's add them when we actually add support for opening model files.

**Comment by stonerl** [comment]
Okay, removed them.

**Review by RunDevelopment — APPROVED** [review]

**Comment by joeyballentine** [comment]
> but double-clicking on them does not do any meaningful for now

Are you sure? On windows if I manually associate the file type, I'm able to open chain files directly by double clicking them. If that doesn't work on Mac, then I guess that means Mac uses different args when launching chaiNNer?

**Comment by joeyballentine** [comment]
Arg parsing: https://github.com/chaiNNer-org/chaiNNer/blob/11a75fcd19e135ce37e7f0d1f2f68025d6305f95/src/main/arguments.ts#L30

Using the arg: https://github.com/chaiNNer-org/chaiNNer/blob/11a75fcd19e135ce37e7f0d1f2f68025d6305f95/src/main/gui/main-window.ts#L129

I believe on windows when you open a file with a program it adds the file as a positional arg. I guess this is not the case for MacOS then

**Comment by stonerl** [comment]
Opening `.chn` files by double-clicking them now also works on macOS.

CleanShot 2023-08-21 at 17 06 14

File: src/public/Info.plist [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            <dict>
+                <key>CFBundleTypeExtensions</key>
+                <array>
+                    <string>chn</string>
+                    <string>CHN</string>
+                </array>
+                <key>CFBundleTypeIconFile</key>
+                <string>file_chn.icns</string>
+                <key>CFBundleTypeName</key>
+                <string>chaiNNer Nodes File</string>
```
[the remark this answers is not in the document]
Comment (stonerl) [reply src/public/Info.plist]: done

**Review by joeyballentine — APPROVED** [review]

**Pull request merged**